src/Microsoft.AspNetCore.Razor.Language/Legacy/CSharpCodeParser.cs

```diff
@@ -993,6 +993,7 @@ private void TypeArgumentList()
             Assert(CSharpSymbolType.LessThan);
             Balance(BalancingModes.None);
             Optional(CSharpSymbolType.GreaterThan);
+            Optional(CSharpSymbolType.QuestionMark);
```

Comment:
This doesn't seem like it should be here. It's not part of a type argument list.